addon-test-support/index.js

```diff
@@ -0,0 +1,28 @@
+import QUnit from 'qunit';
+import { hasRunAllDisposables } from 'ember-lifeline';
+
+const FAILED_ASSERTION_MESSAGE =
+  'One or more objects registered disposables that were not correctly disposed of. Please ensure that objects correctly run their registered disposables by calling `runDisposables` in the `destroy` method of the object.';
+let setupTestDone = false;
+
+export default function setupEnsureLifelineDisposablesRun(hooks) {
+  if (!setupTestDone) {
+    QUnit.testDone(ensureDisposablesRunIfNoFailures);
+
+    setupTestDone = true;
+  }
+
+  hooks.afterEach(ensureDisposablesRun);
+}
+
+export function ensureDisposablesRunIfNoFailures(details) {
+  if (details.failed === 0) {
+    ensureDisposablesRun();
```

Comment:
I think this needs to be a bit different. We can't call `QUnit.assert.ok` in a `testDone`.

Instead, I think it should be something like:

```js
if (!hasRunAllDisposables()) {
  throw new Error(FAILED_ASSERTION_MESSAGE);
}
```

Or something like that...